geo/src/algorithm/intersects.rs

```diff
@@ -211,32 +211,27 @@ where
     T: Float,
 {
     fn intersects(&self, bounding_rect: &Rect<T>) -> bool {
-        // line intersects inner or outer polygon edge
-        if bounding_rect.contains(self) {
-            false
-        } else {
-            let x_overlap = value_in_range(
-                self.min().x,
-                bounding_rect.min().x,
-                bounding_rect.min().x + bounding_rect.width(),
-            ) || value_in_range(
-                bounding_rect.min().x,
-                self.min().x,
-                self.min().x + self.width(),
-            );
-
-            let y_overlap = value_in_range(
-                self.min().y,
-                bounding_rect.min().y,
-                bounding_rect.min().y + bounding_rect.height(),
-            ) || value_in_range(
-                bounding_rect.min().y,
-                self.min().y,
-                self.min().y + self.height(),
-            );
-
-            x_overlap && y_overlap
-        }
+        let x_overlap = value_in_range(
+            self.min().x,
+            bounding_rect.min().x,
+            bounding_rect.max().x
```

Comment:
Simplified this math a bit.

width is computed as  `self.max().x - self.min().x`

So this: 
`self.min().x + self.width()`

previously expanded as:
`self.min().x + self.max().x - self.min().x`

Which is more simply `self.max().x`